router/worker.go

```diff
@@ -258,6 +259,31 @@ func (w *worker) transform(routerJobs []types.RouterJobT) []types.DestinationJob
 		limiterStats.Update(w.partition, time.Since(start), len(routerJobs), 0)
 	}()
 
+	traces := make(map[string]stats.TraceSpan)
+	defer func() {
+		for _, span := range traces {
+			span.End()
+		}
+	}()
+
+	for _, job := range routerJobs {
+		traceParent := job.JobMetadata.TraceParent
+		if traceParent != "" {
+			if _, ok := traces[traceParent]; !ok {
```

Comment:
This deduplication at the time was done to test things out. If we want to keep it then it might make sense to review the tags. For example if the duplication is due to the fact that the same event is due for multiple destinations then it would be confusing to see the span for one destination only and not for the others. Or we can remove the tag and flatten them. Either way I suggest tinkering a bit with the integration test you built to see how is going to look like with 1 source and 3 destinations. Do you see everything as expected?

Reply:
With 1 source connected to 3 destinations, we are getting `rt.pickup` and `rt.process` for each destination.


![Screenshot 2023-12-20 at 4 03 43 PM](https://github.com/rudderlabs/rudder-server/assets/77991816/4e1de121-4615-468f-bc48-ad6766591fdb)
![Screenshot 2023-12-20 at 4 03 41 PM](https://github.com/rudderlabs/rudder-server/assets/77991816/0bbf8ab0-b186-4f8e-a1a4-3676717f246c)
![Screenshot 2023-12-20 at 4 03 35 PM](https://github.com/rudderlabs/rudder-server/assets/77991816/31ea4d5a-0424-4c7b-94a5-6dec891bd893)